crates/router/src/core/authentication.rs

```diff
@@ -257,6 +263,27 @@ pub async fn perform_pre_authentication<F: Clone + Send>(
                 || authentication.authentication_status.is_failed()
             {
                 *should_continue_confirm_transaction = false;
+                // If flow is going through external authentication, set the poll_config in payment_data which can be fetched while sending next_action block in confirm response
+                let default_poll_config = core_types::PollConfig::default();
+                let default_config_str = default_poll_config
+                    .encode_to_string_of_json()
+                    .change_context(errors::ApiErrorResponse::InternalServerError)
+                    .attach_printable("Error while stringifying default poll config")?;
+                let poll_config = state
+                    .store
+                    .find_config_by_key_unwrap_or(
+                        &format!("poll_config_external_three_ds_{authentication_connector_name}"),
+                        Some(default_config_str),
+                    )
+                    .await
+                    .change_context(errors::ApiErrorResponse::InternalServerError)
+                    .attach_printable("The poll config was not found in the DB")?;
+                let poll_config =
+                    serde_json::from_str::<Option<core_types::PollConfig>>(&poll_config.config)
```

Comment:
can you use ParseStruct here? and in other places too where we are doing this for poll config